### ISSUE TITLE
Action does not produce a runnable Node 14 on Mac/Linux/Windows (Node 16 and 18 work fine) when .npmrc includes ignore-scripts=true

### ISSUE THREAD

**Issue by DavidAnson** [actions/setup-node#698]
**Description:**
I'll readily acknowledge this could be MY problem, but I've tried a few things and nothing helps. Briefly, an existing matrix Workflow that worked fine for Node X/Y/Z for years stopped being able to run the `node` executable via `npm run script` for version 14 for all platforms. Links to example workflow below. This reproduces across two different projects.

**Action version:**
v3

**Platform:**
- [x] Ubuntu
- [x] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
Node 14

**Repro steps:**  
Run "CI" workflow in `markdownlint` or `markdownlint-cli2` projects. Existing run linked below.

**Expected behavior:**
"Run All Validations (npm)" script should run successfully for all matrix configurations (which don't special-case anything).

**Actual behavior:**
Nothing happens for Node 14 across all OSes. For diagnostics, I've added a version of that step using "yarn" instead and it also fails, though it emits slightly more information. (FYI, `--scripts-prepend-node-path` did not help.) I've also tried swapping out `npm-run-all` and even invoking `node` directly, but nothing seems to work in the Node 14 scenario.

Example workflow: <https://github.com/DavidAnson/markdownlint/actions/runs/4213895777>
Link to workflow: <https://github.com/DavidAnson/markdownlint/blob/npm-run-nothing/.github/workflows/ci.yml>

I run CI daily for my projects, but because this scenario fails and reports success (ugh), I don't know how long ago it stopped working.

**Comment by DavidAnson** [comment]
Good news, I found the commit on my side that broke this: <https://github.com/DavidAnson/markdownlint/commit/741e6fe72c6549da05e1b2bc9ef6bf07edce5cf7>

Here's the diff in its entirety:

```diff
commit 741e6fe72c6549da05e1b2bc9ef6bf07edce5cf7
Author: David Anson <david@dlaa.me>
Date:   Sun Dec 18 16:01:59 2022 -0800

    Default npm's ignore-scripts flag to true to prevent running pre- or post-scripts in package dependencies.

diff --git a/.npmrc b/.npmrc
index 43c97e7..3757b30 100644
--- a/.npmrc
+++ b/.npmrc
@@ -1 +1,2 @@
+ignore-scripts=true
 package-lock=false
```

I'm not sure I understand why this change should prevent this action from successfully configuring Node 14, though?

**Comment by DavidAnson** [comment]
Confirmed, removing `ignore-scripts=true` from `.npmrc` leads to a runnable Node 14. Example workflow (child of failing workflow in original comment): <https://github.com/DavidAnson/markdownlint/actions/runs/4214042313>

So my questions are:
1. Why does this configuration matter to `setup-node`?
2. Is it a bug in `setup-node`?
3. Can I work around the issue without changing `.npmrc`?

Thanks for your help!

**Comment by MaksimZhukov** [comment]
Hello @DavidAnson !
Thank you for reporting this! We will investigate the issue and answer your questions

**Comment by dmitry-shibanov** [comment]
Hello @DavidAnson. Thank you for your report. I've reproduced the issue with default Nodejs installed from brew that is why I think it's the issue with your set up or Nodejs. 

For now I'm going to close the issue.

**Comment by DavidAnson** [comment]
@dmitry-shibanov @MaksimZhukov Could you please explain why you have closed this issue? As I understand the comment above, you were able to reproduce the problem and believe it is due to the use of brew. To me, that means this project should 1) work around the problem with brew, 2) use a version of brew without the problem, or 3) use something other than brew. There is no requirement for this project to use brew, but there IS an expectation that it produces a runnable version of Node.

We seem to agree that’s not happening here, so why was the issue closed without a fix?

**Comment by dmitry-shibanov** [comment]
Hello @DavidAnson. I've closed the issue because I didn't use the brew with setup-node. I commented setup-node and set up node 14 from brew and could reproduce the issue without using setup-node. The setup-node action does not use brew it downloads archives from nodejs website, extracts it to the hosted toolcache directory and adds it to the PATH.  We could reproduce the issue outside of setup-node action that is why we closed it.
